### PR TITLE
ENH improve config handling

### LINKED ISSUE #102: make adding config options easier (english is hard)
@esheldon had some trouble adding config options to the code because of the current structure which is explicit and quite verbose. We should ease some of this pain by making better use of dictionaries, named tuples, etc.

### PR DESCRIPTION
**Pull request by beckermr**
This PR makes it easier to add configuration options and centralizes some of the bit mask handling that should have already been centralized.

Closes #102

### CODE REVIEW

**Comment by beckermr** [comment]
@esheldon for viz

File: pizza_cutter/des_pizza_cutter/_coadd_slices.py [hunk 1]
```diff
@@ -75,44 +75,67 @@ def _build_slice_inputs(
...
+        region.
     max_unmasked_trail_fraction : float
-        See the documentation of `make_des_pizza_slices` in
-        `pizza_cutter.des_pizz_cutter`.
+        The maximum unmasked bleed trail fraction an SE image can have
+        before it is exlcuded from the coadd. This parameter is the
+        fraction of the subset of the SE image that overlaps the coadd. See
+        the function `compute_unmasked_trail_fraction` in
+        `pizza_cutter.des_pizz_cutter._slice_flagging.` for details.
     mask_tape_bumps: boold
```
Comment (esheldon) [inline pizza_cutter/des_pizza_cutter/_coadd_slices.py]: may as well fix this typo while we are at it

Comment (beckermr) [reply pizza_cutter/des_pizza_cutter/_coadd_slices.py]: thanks for fixing!

**Comment by beckermr** [comment]
OK to merge @esheldon?

**Comment by esheldon** [comment]
I think so

**Pull request merged**